clients/spark/core/src/main/scala/io/treeverse/clients/Exporter.scala

```diff
@@ -102,7 +113,7 @@ class Exporter(spark : SparkSession, apiClient: ApiClient, filter: KeyFilter, re
 
   private def writeSummaryFile(success: Boolean, commitID: String, content : String) = {
     val suffix = if(success) "SUCCESS" else "FAILURE"
-    val dstPath = resolveURL(new URI(dstRoot), s"EXPORT_${commitID}_${suffix}")
+    val dstPath = URLResolver.resolveURL(new URI(dstRoot), s"EXPORT_${commitID}_${suffix}")
```

Comment:
Why do we need this change?

Reply:
`Handler` uses it and calling `resolveURL`. Since `resolveURL` was part of `Exporter` it had to be `Serializable`. It's easier to use another struct than making our `APIClient` and other members `Serializable` too.

Reply:
Not sure I follow.  `resolveURL` is a method, Spark tries to serialize a method?!

Reply:
Nope, if it is called inside a spark task (`exporter.resolveURL(...)`), the entire `exporter` class will be passed to the task (hence needs to be `Serializable`) to use its method. Even if `resolveURL` uses none of the `exporter` properties. 
That's why it made much more sense to extract that single method to its own simple class.